### ISSUE TITLE
Alpha version 3.0.0 not uploading files and crashing

### ISSUE THREAD

**Issue by inis17** [haiwen/seadroid#1001]
When trying to upload files, with the alpha version 3.0.0 files won't upload.
##### System
Pixel 7a, 
Graphen OS.
apk obtain via obtainium on haiwen/seadroid

##### Step to reproduce : 
i tried to upload a file in one of my library : 
<img src="https://github.com/haiwen/seadroid/assets/56741837/c4063766-acc7-4057-a258-7c32a4c99e60" width="400">

Then i receive this notification. (i must say that i don't have storage limit on my account and the drive is far from full)
<img src="https://github.com/haiwen/seadroid/assets/56741837/52f0856e-d0ea-4e5e-b58d-e1ae401ed02e" width="400">

Then if i repeat this steps on a password protected library the app crash ans won't start for a moment (1-2 min)
<img src="https://github.com/haiwen/seadroid/assets/56741837/69cb4a12-feea-4d2b-908a-a39e90fa174a" width="400">

You can find the crash report below.
[Error in Seafile 5aeecd25e62b.txt](https://github.com/haiwen/seadroid/files/15203451/Error.in.Seafile.5aeecd25e62b.txt)

Thank you for seafile and the work you're doing !

**Comment by zhwanng** [comment]
Thank you for your feedback. I will continue to track it

**Comment by TuxCoder** [comment]
I have the same bug,
same settings (unlimited quota)

If I understand correctly:
here you parse the account info from a user:
https://github.com/haiwen/seadroid/blob/master/app/src/main/java/com/seafile/seadroid2/account/AccountInfo.java#L35

On a webbrowser I geht this data for it `/api2/account/info/`:
```json
{
   "avatar_url" : "https://seafile.XXX/media/avatars/default.png",
   "collaborate_email_interval" : 0,
   "contact_email" : null,
   "department" : "",
   "email" : "tuxcoder@XXX",
   "file_updates_email_interval" : 0,
   "institution" : "",
   "is_staff" : true,
   "login_id" : "",
   "name" : "XXX",
   "space_usage" : "0%",
   "total" : -2,
   "usage" : 62962239976
}
```

here the quota is calculated but not take in account, that `total` - value could be negative (eg. unlimited)
https://github.com/haiwen/seadroid/blob/master/app/src/main/java/com/seafile/seadroid2/framework/worker/BaseUploadFileWorker.java#L68


Think the right patch would to add in the `calculateQuota` function add a case if total is negative return true

**Comment by TuxCoder** [comment]
Also set my account quota to something high (100GB) but not unlimited and it works, so it has clearly something to do with the quota.

Also here Server data:
Using seafile package from nixos : https://search.nixos.org/packages?channel=23.11&from=0&size=50&sort=relevance&type=packages&query=seafile-server
version 10.0.1

**Comment by zhwanng** [comment]
yes, please wait a fixed version.

**Comment by zhwanng** [comment]
please update to v3.0.1-beta and try it again

**Comment by notdeadyett** [comment]
This is still an issue with v3.0.1-beta

**Comment by zhwanng** [comment]
Still showing "Out of quota" and "Crashing"? 
can describe it in more detail? 
or provide a screenshot or log file?

**Comment by notdeadyett** [comment]
![Screenshot_20240704-222057_1](https://github.com/haiwen/seadroid/assets/35095924/06a97894-b4ff-428d-878d-d267819ed197)

I get the Above Quota error when uploading any file, however the app doesn't crash, it just doesn't actually upload the file.

I'm not sure how to get logs from the app.

Thanks

**Comment by zhwanng** [comment]
Not quite pinpointing the cause of the problem yet, is it possible to answer the following questions?

1. what is the used space quota and total quota? or unlimited?
2. manual file upload or folder backup feature?

tips:
The directory for the log files is at: /sdcard/Android/data/com.seafile.seadroid2/files/logs/

Or provide json data like this，e.g. [the answer above](https://github.com/haiwen/seadroid/issues/1001#issuecomment-2117118699)

**Comment by notdeadyett** [comment]
Manual upload, I haven't tried folder backup yet.

Here is the json data:

`{
  "space_usage": "0%",
  "avatar_url": "https://seafile.XXX",
  "email": "XXX",
  "name": "notdeadyett",
  "total": -2,
  "usage": 89641543605,
  "login_id": "",
  "department": "",
  "contact_email": "XXX",
  "institution": "",
  "is_staff": true,
  "enable_subscription": false,
  "file_updates_email_interval": 0,
  "collaborate_email_interval": 0
}`

**Comment by Evangelospro** [comment]
Seems to be fixed for me with `v3.0.2-pre`. I think this issue can be closed!
